llmfoundry/eval/datasets/in_context_learning_evaluation.py

```diff
@@ -0,0 +1,1939 @@
+# Copyright 2024 MosaicML LLM Foundry authors
+# SPDX-License-Identifier: Apache-2.0
+
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+# This code is based on the implementation in https://github.com/EleutherAI/lm-evaluation-harness/blob/8c048e266a22a1c85ccbdb0c209ac712e4f39989/lm_eval/base.py#L221-L330
+
+from __future__ import annotations
+
+import copy
+import json
+import os
+import random
+import warnings
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Set, Union
+
+import torch
+from composer.core import DataSpec
+from composer.core.data_spec import _default_split_batch, _split_list
+from composer.datasets.utils import stop_sequences_criteria
+from composer.utils import MissingConditionalImportError, dist, get_file
+from torch.utils.data import DataLoader, Dataset
+
+if TYPE_CHECKING:
+    import transformers
+    from datasets import \
+        Dataset as HFDataset  # pyright: ignore[reportGeneralTypeIssues]
+
+# Allow models to have slightly more tokens than were used in the most verbose CoT in the dataset
+_MAX_ANSWER_BUFFER_LENGTH = 10
+
+__all__ = [
+    'InContextLearningLMTaskDataset',
+    'InContextLearningMultipleChoiceTaskDataset',
+    'InContextLearningCodeEvalDataset',
+    'InContextLearningQATaskDataset',
+    'get_icl_task_dataloader',
+]
+
+
+def strip_data(example: Dict) -> Dict:
```

Comment:
I can see an argument to move these to a utils file rather than leave them in the `in_context_learning_evaluation.py` file